### PR TITLE
Update ftc branch to FTC SDK release v16.02.09.001

### PR DESCRIPTION
**Pull request by lizlooney**
Changed the behavior when a hardware component's DeviceName is not found in the
configuration. Previous to this change, it was a fatal error if an app had a
hardware component whose DeviceName was not found in the configuration and the
fatal error occurred just before the OpMode's Init event was triggered, even if
the OpMode didn't use that hardware component. With this change, the fatal error
will occur only when an OpMode actually tries to use a hardware component whose
DeviceName is not found in the configuration.

If an app has no FtcRobotController component, the Form will add the Exit menu
item to the options menu.

Added FtcGamepad Type property getter.

Added FtcRobotController.TelemetrySorted (property getter & setter).

Make designer change a FtcGamepad components's GamepadNumber property to "2" if
the name is FtcGamepad2 (localized).

Modified FtcHardwareDevice.initHardwareDevice() to check explicitly for hardware
being null rather than calling toString().

Corrected spelling of 'Analytics' in FTC LICENSE file.

Minor changes to block descriptions.

Updated to FTC SDK release v16.02.09.001.

### CODE REVIEW

**Comment by lizlooney** [comment]
@jisqyv

File: appinventor/components/src/com/google/appinventor/components/common/YaVersion.java [hunk 1]
```diff
@@ -411,6 +411,8 @@ private YaVersion() {
   // - FTC_DC_MOTOR_CONTROLLER_COMPONENT_VERSION was incremented to 3.
   // For YOUNG_ANDROID_VERSION 145:
   // - Spelling of "obsfucate" was corrected to obfuscate and BLOCKS_LANGUAGE_VERSION incremented to 20
+  // - FTC_GAMEPAD_COMPONENT_VERSION was incremented to 2.
+  // - FTC_ROBOT_CONTROLLER_COMPONENT_VERSION was incremented to 3.
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/common/YaVersion.java]: @lizlooney Is this safe? Or should bump YaVersion (and reserve it in the master branch)?

**Comment by jisqyv** [comment]
@lizlooney Has this code been tested?

File: appinventor/components/src/com/google/appinventor/components/common/YaVersion.java (hunk 1 above)
Comment (lizlooney) [inline appinventor/components/src/com/google/appinventor/components/common/YaVersion.java]: Yes, it's safe. I only need to bump the version number if I add/change a component and the master hasn't needed to bump the version number. In this case, these changes were added a month ago, but I hadn't merged in the changes from master, so they were added at 145.

**Comment by lizlooney** [comment]
This exact combination has not been tested, but I think that's fine. This pull request merges in the current beta release for FTC, which has been well tested.

FTC makes their beta and official releases from my github repo (https://github.com/lizlooney/appinventor-sources/releases). The current beta release was made on 2/19 (before your most recent merge from master into ftc). It will be promoted to "official" on Wednesday 3/9.

I will be making another release on Wednesday 3/9 or Thursday 3/10 (depending on when FIRST & Qualcomm finish their part), which will become the next beta release for FTC.

**Comment by jisqyv** [comment]
@lizlooney Should I merge this now, or wait until Thursday?

**Comment by lizlooney** [comment]
I think you should merge it now. There will be some more changes for the next release, but I won't push them until I get the official release from FIRST/Qualcomm.

**Comment by lizlooney** [comment]
Thanks!

**Comment by jisqyv** [comment]
@lizlooney MERGED

**Pull request merged**